Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,13 +517,70 @@ bool isFunctionJson(string fullName)
             }
         }
 
-        private async Task<string> DeployZipLocally(ZipDeploymentInfo zipDeploymentInfo, ITracer tracer)
+        // OneDeploy Fetch handler for non-zip artifacts.
+        // For zip files, OneDeploy uses the LocalZipHandler Fetch handler
+        // NOTE: Do not access the request stream as it may have been closed during asynchronous scenarios
+        private async Task OneDeployFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
         {
-            var content = await DeploymentHelper.GetZipContentFromURL(zipDeploymentInfo, tracer);
+            var artifactDeploymentInfo = (ArtifactDeploymentInfo) deploymentInfo;
+
+            // This is the path where the artifact being deployed is staged, before it is copied to the final target location
+            var artifactDirectoryStagingPath = repository.RepositoryPath;
+
+            var targetInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(artifactDirectoryStagingPath);
+            if (targetInfo.Exists)
+            {
+                // If tempDirPath already exists, rename it so we can delete it later 
+                var moveTarget = Path.Combine(targetInfo.Parent.FullName, Path.GetRandomFileName());
+                using (tracer.Step(string.Format("Renaming ({0}) to ({1})", targetInfo.FullName, moveTarget)))
+                {
+                    targetInfo.MoveTo(moveTarget);
+                }
+            }
+
+            // Create artifact staging directory before later use 
+            Directory.CreateDirectory(artifactDirectoryStagingPath);
+            var artifactFileStagingPath = Path.Combine(artifactDirectoryStagingPath, deploymentInfo.TargetFileName);
+
+            // If RemoteUrl is non-null, it means the content needs to be downloaded from the Url source to the staging location
+            // Else, it had been downloaded already so we just move it to the staging location
```

Comment:
// Else, it had been downloaded already so we just move the downloaded file to the staging location